### PR TITLE
feat(cdk): support provide custom query selector for auto focus directive

### LINKED ISSUE #9092: 🐞 - [v4] `AutoFocus` does not work with `tui-editor`
### Code Example URL

https://github.com/taiga-family/taiga-ui/blob/5ce088ab2becd92062f3f4ef94a0bf0a8373c975/projects/cdk/directives/auto-focus/handlers/abstract.handler.ts#L13

### Description

`AutoFocus` directive does not find tui-editor's `[contenteditable]` element because of the CSS selector issue here:
https://github.com/taiga-family/taiga-ui/blob/5ce088ab2becd92062f3f4ef94a0bf0a8373c975/projects/cdk/directives/auto-focus/handlers/abstract.handler.ts#L13

Used to work in `v3`.

### Taiga UI version

4.7.0

### Browser Used

Chrome

### OS Used

Windows

### PR DESCRIPTION
**Pull request by splincode**
image

Close #9092

### CODE REVIEW

**Review by mdlufy — APPROVED** [review]

File: projects/cdk/directives/auto-focus/autofocus.options.ts [hunk 1]
```diff
@@ -14,10 +14,12 @@ export interface TuiAutofocusHandler {
 
 export interface TuiAutofocusOptions {
     readonly delay: number;
+    readonly query: string;
 }
 
 export const TUI_AUTOFOCUS_DEFAULT_OPTIONS: TuiAutofocusOptions = {
     delay: NaN, // NaN = no delay/sync
+    query: 'input,textarea',
```
Comment (hakimio) [inline projects/cdk/directives/auto-focus/autofocus.options.ts:22]: Default should be `input, textarea, [contenteditable]`.

**Review by waterplea — APPROVED** [review]

File: projects/cdk/directives/auto-focus/autofocus.options.ts (hunk 1 above)
Comment (waterplea) [reply projects/cdk/directives/auto-focus/autofocus.options.ts:22]: Probably also `, select`

File: projects/cdk/directives/auto-focus/handlers/abstract.handler.ts [hunk 2]
```diff
@@ -1,16 +1,19 @@
...
+        protected readonly options: TuiAutofocusOptions,
+    ) {}
 
     public abstract setFocus(): void;
 
     protected get element(): HTMLElement {
         // TODO: Remove when legacy controls are dropped
         const el = this.el.nativeElement.tagName.includes('-')
-            ? this.el.nativeElement.querySelector<HTMLElement>('input,textarea')
+            ? this.el.nativeElement.querySelector<HTMLElement>(this.options.query)
```
Comment (hakimio) [inline projects/cdk/directives/auto-focus/handlers/abstract.handler.ts:16]: Even if you specify custom query here, `isTextFieldElement()` check would prevent it from working.

File: projects/cdk/directives/auto-focus/autofocus.options.ts (hunk 1 above)
Comment (splincode) [reply projects/cdk/directives/auto-focus/autofocus.options.ts]: fixed

File: projects/cdk/directives/auto-focus/handlers/abstract.handler.ts (hunk 2 above)
Comment (splincode) [reply projects/cdk/directives/auto-focus/handlers/abstract.handler.ts:16]: fixed

**Review by hakimio — APPROVED** [review]

**Review by hakimio — APPROVED** [review]

**Review by waterplea — APPROVED** [review]

**Pull request merged**